sklearn/_loss/_loss.pyx.tp

```diff
@@ -765,9 +765,8 @@ cdef inline double_pair cgrad_hess_half_binomial(
         gh.val1 = ((1 - y_true) - y_true * gh.val2) / (1 + gh.val2)  # gradient
         gh.val2 = gh.val2 / (1 + gh.val2)**2                         # hessian
     else:
-        gh.val2 = exp(raw_prediction)
+        gh.val2 = exp(raw_prediction)  # = 1. order Taylor in exp(raw_prediction)
```

Comment:
We aren't correct the formula here right? Because I wasn't able to relate the changes with the correct formulas that you have presented.

Reply:
The statement `gh.val2 *= (1 - gh.val2)` just below should read `gh.val2 *= (1 - 2 * gh.val2)`. But it is completely unimportant as this is a second order term of the order or `1e-16 ** 2`. So I just removed that statement completely, `gh.val2 = exp(raw_prediction)` (hessian) is already the first order term.

Reply:
Okay got it thanks for clarifying.